### DISCUSSION TITLE
WAF plugin

### DISCUSSION THREAD

**Discussion by ahmadnassri** [Kong/kong#7397] [Ideas and feature requests]
[naxsi](https://github.com/nbs-system/naxsi) provides a [WAF](https://www.owasp.org/index.php/Web_Application_Firewall) layer for nginx

might be suited for a plugin.

**Comment by Tieske** [comment]
from that site;

> The effort to perform this customization can be significant and needs to be maintained as the application is modified

sounds like an application specific plugin, not something generic.

**Comment by ahmadnassri** [comment]
related: https://www.trustwave.com/Resources/SpiderLabs-Blog/Announcing-the-availability-of-ModSecurity-extension-for-Nginx/

**Comment by thibaultcha** [comment]
https://github.com/p0pr0ck5/FreeWAF

**Comment by Tieske** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;6ydwaD3t&#x2F;2-waf-plugin"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> WAF plugin</a>

**Comment by tuler** [comment]
Is this trello board public?

**Comment by Tieske** [comment]
nope, it isn't. my mistake. Was playing around with Trello.

**Comment by mwisner** [comment]
Has anyone experimented with kong+https://github.com/p0pr0ck5/lua-resty-waf? I believe it's listed above but now has a new name a what appears to be some significant changes.

**Comment by samrose** [comment]
I got this set up as a separate entity: https://github.com/p0pr0ck5/lua-resty-waf the developer is very helpful too. 

I also did compile it once with Kong, although I did not do extensive testing of it.

**Comment by jkumar-roambee** [comment]
+1

**Comment by esatterwhite** [comment]
What happened with this? is there any traction here?

**Comment by samrose** [comment]
I am noting this update on `lua-resty-waf` which states https://github.com/p0pr0ck5/lua-resty-waf#status

> NOTE: lua-resty-waf is essentially abandoned. This project had a use in a time when ModSecurity for Nginx was not a viable option; this is no longer the case. There was an attempt to revitalize the project in 2020, but I do not have the resources to complete this; this work is partially complete in the redux branch.

Lately it seems the consensus is moving toward https://discuss.konghq.com/t/modsec-v3-owasp-crs3-on-kong-discussion-guide/5402

Kong + ModeSecv3 + OWASP CRS
